Modules/_zoneinfo.c

```diff
@@ -2189,8 +2197,14 @@ find_ttinfo(zoneinfo_state *state, PyZoneInfo_ZoneInfo *self, PyObject *dt)
         }
     }
 
+    PyDateTime_DateTime *local_dt = as_local_datetime(dt);
+    if (local_dt == NULL) {
+        return NULL;
+    }
     int64_t ts;
-    if (get_local_timestamp(dt, &ts)) {
+    int failed = get_local_timestamp(local_dt, &ts);
+    Py_DECREF(local_dt);
+    if (failed) {
         return NULL;
     }
 
```

Comment:
Make sense, updated. Thanks for the suggestion!

Reply:
I like the idea, but I think there's an type issue at `get_local_timestamp(dt, &ts)` when I tried it

Reply:
You could either pass `local_dt` (which is the same pointer but a different type), or cast `dt` with `(PyDateTime_DateTime *)dt)` since you know it's safe by construction.

Reply:
Thanks for the suggestion! I suppose this will also use the object without timezone info, which I'm not sure is desired here..